lib/app/modules/smart/smart_discovery_sequence.rb

```diff
@@ -10,129 +10,216 @@ class SMARTDiscoverySequence < SequenceBase
+        assert_response_ok(well_known_configuration_response)
+        assert_response_content_type(well_known_configuration_response, 'application/json')
+        assert_valid_json(well_known_configuration_response.body)
+
+        @well_known_configuration = JSON.parse(well_known_configuration_response.body)
+        @well_known_authorize_url = @well_known_configuration['authorization_endpoint']
+        @well_known_token_url = @well_known_configuration['token_endpoint']
+        @well_known_register_url = @well_known_configuration['registration_endpoint']
+        @well_known_manage_url = @well_known_configuration['management_endpoint']
+        @well_known_introspect_url = @well_known_configuration['introspection_endpoint']
+        @well_known_revoke_url = @well_known_configuration['revocation_endpoint']
+
+        @instance.update(
+          oauth_authorize_endpoint: @well_known_authorize_url,
+          oauth_token_endpoint: @well_known_token_url,
+          oauth_register_endpoint: @well_known_configuration['registration_endpoint']
+        )
+
+        assert @well_known_configuration.present?, 'No .well-known/smart-configuration body'
+      end
 
-        @instance.update(oauth_authorize_endpoint: @well_known_authorize_url, oauth_token_endpoint: @well_known_token_url, oauth_register_endpoint: register_url)
+      test 'Configuration from well-known endpoint contains required fields' do
+        metadata do
+          id '02'
+          link 'http://hl7.org/fhir/smart-app-launch/conformance/index.html#metadata'
+          desc %(
+            The JSON from .well-known/smart-configuration contains the following
+            required fields: #{REQUIRED_WELL_KNOWN_FIELDS.map { |field| "`#{field}`" }.join(', ')}
+          )
+        end
 
-        assert !@well_known_authorize_url.blank?, 'No authorize URI provided in response.'
-        assert !@well_known_token_url.blank?, 'No token URI provided in response.'
-        assert !capabilities.nil?, 'The response did not contain capabilities as required'
-        assert capabilities.is_a?(Array), 'The capabilities response is not an array'
+        missing_fields = REQUIRED_WELL_KNOWN_FIELDS - @well_known_configuration.keys
+        assert missing_fields.empty?, "The following required fields are missing: #{missing_fields.join(', ')}"
       end
 
-      test 'Conformance/Capability Statement provides OAuth 2.0 endpoints' do
+      test 'Configuration from well-known endpoint contains recommended fields' do
         metadata do
-          id '02'
-          link 'http://www.hl7.org/fhir/smart-app-launch/capability-statement/'
+          id '03'
+          link 'http://hl7.org/fhir/smart-app-launch/conformance/index.html#metadata'
+          optional
           desc %(
+            The JSON from .well-known/smart-configuration contains the following
+            recommended fields: #{RECOMMENDED_WELL_KNOWN_FIELDS.map { |field| "`#{field}`" }.join(', ')}
```

Comment:
```suggestion
            recommended fields: #{RECOMMENDED_WELL_KNOWN_FIELDS.map { |field| "`#{field}`" }.join(', ')}.
            
            This test is optional because these fields are recommended, not required.
```